Source/com/drew/imaging/png/PngMetadataReader.java

```diff
@@ -323,6 +327,21 @@ private static void processChunk(@NotNull Metadata metadata, @NotNull PngChunk c
             PngDirectory directory = new PngDirectory(PngChunkType.sBIT);
             directory.setByteArray(PngDirectory.TAG_SIGNIFICANT_BITS, bytes);
             metadata.addDirectory(directory);
+        } else if (chunkType.equals(PngChunkType.eXIF)) {
+            try {
+                ExifTiffHandler handler = new ExifTiffHandler(metadata, null);
+                RandomAccessStreamReader reader = new RandomAccessStreamReader(new ByteArrayInputStream(bytes));
+                int tiffHeaderOffset = 0;
+                new TiffReader().processTiff(reader, handler, tiffHeaderOffset);
```

Comment:
This can avoid a layer of abstraction.

```suggestion
                new TiffReader().processTiff(new ByteArrayReader(bytes), handler, 0);
```